sidekick_core/lib/src/dart_package.dart

```diff
@@ -28,8 +28,8 @@ class DartPackage {
 
       // Check for (optional) flutter dependency
       final deps = pubspec['dependencies'] as YamlMap?;
-      final flutterDep = deps?['flutter'] as YamlMap?;
-      if (flutterDep != null) {
+      final withFlutter = deps?.containsKey('flutter') == true;
```

Comment:
Just a nit, feel free to ignore
```suggestion
      final withFlutter = deps?.containsKey('flutter') ?? false;
```